services/provider/server/server.go

```diff
@@ -610,358 +608,172 @@ func decodeAndValidateTicket(ticket string, pubKey *rsa.PublicKey) (*services.On
-		klog.Error(errMsg)
-		if kerrors.IsAlreadyExists(err) {
-			return nil, status.Error(codes.AlreadyExists, errMsg)
-		}
-		return nil, status.Error(codes.Internal, errMsg)
-	}
-
-	return &pb.FulfillStorageClaimResponse{}, nil
+	return nil, nil
 }
 
 // RevokeStorageClaim RPC call to delete the StorageClaim CR on
-// provider cluster.
+// provider cluster is no longer supported since 4.19 so giving failure.
 func (s *OCSProviderServer) RevokeStorageClaim(ctx context.Context, req *pb.RevokeStorageClaimRequest) (*pb.RevokeStorageClaimResponse, error) {
-	err := s.storageRequestManager.Delete(ctx, req.StorageConsumerUUID, req.StorageClaimName)
-	if err != nil {
-		errMsg := fmt.Sprintf("failed to revoke storage class claim %q for %q. %v", req.StorageClaimName, req.StorageConsumerUUID, err)
-		klog.Error(errMsg)
-		return nil, status.Error(codes.Internal, errMsg)
-	}
-
-	return &pb.RevokeStorageClaimResponse{}, nil
-}
-
-func storageClaimCephCsiSecretName(secretType, suffix string) string {
-	return fmt.Sprintf("ceph-client-%s-%s", secretType, suffix)
+	errMsg := fmt.Sprintf("failed to revoke storage class claim %q for %q.", req.StorageClaimName, req.StorageConsumerUUID)
+	klog.Error(errMsg)
+	return nil, status.Error(codes.Internal, errMsg)
 }
 
 // GetStorageClaim RPC call to get the ceph resources for the StorageClaim.
 func (s *OCSProviderServer) GetStorageClaimConfig(ctx context.Context, req *pb.StorageClaimConfigRequest) (*pb.StorageClaimConfigResponse, error) {
-	storageRequest, err := s.storageRequestManager.Get(ctx, req.StorageConsumerUUID, req.StorageClaimName)
+	storageConsumer, err := s.consumerManager.Get(ctx, req.StorageConsumerUUID)
 	if err != nil {
-		errMsg := fmt.Sprintf("failed to get storage class claim config %q for %q. %v", req.StorageClaimName, req.StorageConsumerUUID, err)
-		if kerrors.IsNotFound(err) {
-			return nil, status.Error(codes.NotFound, errMsg)
-		}
-		return nil, status.Error(codes.Internal, errMsg)
-	}
-
-	// Verify Status.Phase
-	msg := fmt.Sprintf("storage claim %q for %q is in %q phase", req.StorageClaimName, req.StorageConsumerUUID, storageRequest.Status.Phase)
-	klog.Info(msg)
-	if storageRequest.Status.Phase != ocsv1alpha1.StorageRequestReady {
-		switch storageRequest.Status.Phase {
-		case ocsv1alpha1.StorageRequestFailed:
-			return nil, status.Error(codes.Internal, msg)
-		case ocsv1alpha1.StorageRequestInitializing:
-			return nil, status.Error(codes.Unavailable, msg)
-		case ocsv1alpha1.StorageRequestCreating:
-			return nil, status.Error(codes.Unavailable, msg)
-		case "":
-			return nil, status.Errorf(codes.Unavailable, "status is not set for storage class claim %q for %q", req.StorageClaimName, req.StorageConsumerUUID)
-		default:
-			return nil, status.Error(codes.Internal, msg)
-		}
+		return nil, err
+	}
+	storageCluster, err := util.GetStorageClusterInNamespace(ctx, s.client, s.namespace)
+	if err != nil {
+		return nil, err
 	}
```

Comment:
Are we using the storageCluster anywhere?

Reply:
Yes, `req.StorageClaimName == util.GenerateNameForCephBlockPoolStorageClass(storageCluster)`